configs/projects/puppet-agent.rb

```diff
@@ -30,6 +30,18 @@
 
   platform = proj.get_platform
 
+  # HuaweiOS is a cross-compiled platform
+  if platform.is_huaweios?
+    platform_triple = "powerpc-unknown-linux-gnu"
+    host = "--host #{platform_triple}"
+
+    # Use a standalone ruby for cross-compilation
+    proj.setting(:host_ruby, "/opt/pl-build-tools/bin/ruby")
+    proj.setting(:host_gem, "/opt/pl-build-tools/bin/gem")
+
+    proj.noarch
```

Comment:
Is it really noarch? Don't we want to confine architecture? https://github.com/puppetlabs/vanagon/blob/5236eefe97963f3cfc312bbef1e3e1b388d6ffc4/resources/deb/control.erb#L10